datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -302,14 +302,20 @@ def _request(self, method, url, options):
         if persist is None:
             persist = self.persist_connections
 
+        new_options = self.populate_options(options)
```

Comment:
We only use this in one place, can we just avoid assignment like before? or at least move it closer to where it's used...

Reply:
Well, the `new_options['headers']` needs to be updated at line `new_options['headers'].update(extra_headers)`.

I you prefer, I can move those few new lines inside the `with ExitStack() as stack:` block.

Reply:
Ah my bad